plugins/modules/rma_workflow_manager.py

```diff
@@ -0,0 +1,987 @@
+            - Updates the status, msg, and result attributes based on the task result.
+            - Handles any exceptions that occur during the process.
+        """
+
+        # Check if faulty device ID is available
+        if not self.have.get("faulty_device_id"):
+            self.log("Faulty device ID is missing", "ERROR")
+            self.status = "failed"
+            self.msg = "Faulty device ID is missing"
+            return self
+
+        import_params = dict(
+            payload=[{
+                "faultyDeviceId": self.have.get("faulty_device_id"),
+                "replacementStatus": "MARKED-FOR-REPLACEMENT"
+            }],
+        )
+
+        try:
+            response = self.dnac._exec(
+                family="device_replacement",
+                function='unmark_device_for_replacement',
+                op_modifies=True,
+                params=import_params
+            )
+            self.log("Received API response from 'unmark_device_for_replacement': {0}".format(self.pprint(response)), "DEBUG")
+            task_id = response.get("response", {}).get("taskId")
+            task_result = self.check_rma_task_status(
+                task_id,
+                "Device unmarked for replacement successfully",
+                "Error while unmarking device for replacement"
+            )
+            self.status = task_result["status"]
+            self.msg = task_result["msg"]
+            if self.status == "success":
+                self.result['changed'] = True
+
+        except Exception as e:
+            self.status = "failed"
+            self.msg = "Exception occurred while unmarking device for replacement: {0}".format(str(e))
+            self.log(self.msg, "ERROR")
+        return self
+
+    def check_rma_task_status(self, task_id, success_message, error_prefix):
+        """
+        Check the status of an RMA task in Cisco Catalyst Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            task_id (str): The ID of the task to monitor.
+            success_message (str): The message to log on successful completion.
+            error_prefix (str): The prefix for the error message if the task fails.
+        Returns:
+            dict: A dictionary containing the status and message of the task result.
+        Description:
+            This method checks the status of an RMA task in Cisco Catalyst Center. It performs the following steps:
+            - Continuously polls the task status using the get_task_details method.
+            - Checks if the task has completed successfully or encountered an error.
+            - Logs appropriate messages based on the task outcome.
+            - Returns a dictionary with the task status and message.
+            - Implements a delay between status checks to avoid overwhelming the API.
+        """
+
+        dnac_poll_interval = 2
+        timeout_inreval = 100
+        while True:

```

Comment:
We are checking  if not task_details.get("isError") two times.. Also if we write failed condition first, we will return.. If successful in task then we return success. Otherwise we will decrement the timeout_interval..

```
   while timeout_interval > 0:
        task_details = self.get_task_details(task_id)

        if task_details.get("isError"):
            error_message = task_details.get("failureReason", "{0}: Task failed.".format(error_prefix))
            self.log(error_message, "ERROR")
            return {"status": "failed", "msg": error_message}
        
        if 'successful' in task_details.get("progress", ""):
            self.log(success_message, "INFO")
            return {"status": "success", "msg": task_details.get("progress")}
        
        time.sleep(ccc_poll_interval)
        timeout_interval -= ccc_poll_interval
```